### ISSUE TITLE
Global Step Local Train

### ISSUE THREAD

**Issue by mshearer0** [GoogleCloudPlatform/data-science-on-gcp#11]
train_local runs 2000 steps not 200, if fix_global_step_increment_bug=True is defined then 1000 steps.

batch size =512
train.csv has 10003 rows

20 batches x 10 epochs?

WARNING:tensorflow:The default stddev value of initializer will change from "1/sqrt(vocab_size)" to "1/sqrt(dimension)" after 2017/02/25.
WARNING:tensorflow:The default stddev value of initializer will change from "1/sqrt(vocab_size)" to "1/sqrt(dimension)" after 2017/02/25.
WARNING:tensorflow:The default stddev value of initializer will change from "1/sqrt(vocab_size)" to "1/sqrt(dimension)" after 2017/02/25.
WARNING:tensorflow:The default stddev value of initializer will change from "1/sqrt(vocab_size)" to "1/sqrt(dimension)" after 2017/02/25.
WARNING:tensorflow:The default stddev value of initializer will change from "1/sqrt(vocab_size)" to "1/sqrt(dimension)" after 2017/02/25.
WARNING:tensorflow:The default stddev value of initializer will change from "1/sqrt(vocab_size)" to "1/sqrt(dimension)" after 2017/02/25.
WARNING:tensorflow:The default stddev value of initializer will change from "1/sqrt(vocab_size)" to "1/sqrt(dimension)" after 2017/02/25.
INFO:tensorflow:Using default config.
INFO:tensorflow:Using config: {'_save_checkpoints_secs': 600, '_num_ps_replicas': 0, '_keep_checkpoint_max': 5, '_task_type': None, '_is_chief': True, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f3c87b04890>, '_model_dir': './trained_model/', '_save_checkpoints_steps': None, '_keep_checkpoint_every_n_hours': 10000, '_session_config': None, '_tf_random_seed': None, '_save_summary_steps': 100, '_environment': 'local', '_num_worker_replicas': 0, '_task_id': 0, '_log_step_count_steps': 100, '_tf_config': gpu_options {
  per_process_gpu_memory_fraction: 1.0
}
, '_evaluation_master': '', '_master': ''}
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/learn/python/learn/monitors.py:267: __init__ (from tensorflow.contrib.learn.python.learn.monitors) is deprecated and will be removed after 2016-12-05.
Instructions for updating:
Monitors are deprecated. Please use tf.train.SessionRunHook.
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/contrib/layers/python/layers/feature_column.py:2341: calling sparse_feature_cross (from tensorflow.contrib.layers.python.ops.sparse_feature_cross_op) with hash_key=None is deprecated and will be removed after 2016-11-20.
Instructions for updating:
The default behavior of sparse_feature_cross is changing, the default
value for hash_key will change to SPARSE_FEATURE_CROSS_DEFAULT_HASH_KEY.
From that point on sparse_feature_cross will always use FingerprintCat64
to concatenate the feature fingerprints. And the underlying
_sparse_feature_cross_op.sparse_feature_cross operation will be marked
as deprecated.
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 1 into ./trained_model/model.ckpt.
INFO:tensorflow:loss = 9.615792, step = 1
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
INFO:tensorflow:Starting evaluation at 2018-04-01-14:20:46
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from ./trained_model/model.ckpt-1
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Evaluation [10/100]
INFO:tensorflow:Evaluation [20/100]
INFO:tensorflow:Finished evaluation at 2018-04-01-14:20:48
INFO:tensorflow:Saving dict for global step 1: accuracy = 0.16655004, accuracy/baseline_label_mean = 0.83344996, accuracy/threshold_0.700000_mean = 0.16655004, auc = 0.5, auc_precision_recall = 0.916725, global_step = 1, labels/actual_label_mean = 0.83344996, labels/prediction_mean = 2.4281659e-22, loss = 108.89304, precision/positive_threshold_0.700000_mean = 0.0, recall/positive_threshold_0.700000_mean = 0.0, rmse = 0.9129348, training/hptuning/metric = 0.9129348
INFO:tensorflow:Validation (step 100): accuracy/baseline_label_mean = 0.83344996, loss = 108.89304, auc = 0.5, accuracy/threshold_0.700000_mean = 0.16655004, global_step = 1, rmse = 0.9129348, recall/positive_threshold_0.700000_mean = 0.0, labels/prediction_mean = 2.4281659e-22, precision/positive_threshold_0.700000_mean = 0.0, training/hptuning/metric = 0.9129348, accuracy = 0.16655004, auc_precision_recall = 0.916725, labels/actual_label_mean = 0.83344996
INFO:tensorflow:global_step/sec: 16.7229
INFO:tensorflow:loss = 0.07211667, step = 101 (5.980 sec)
INFO:tensorflow:global_step/sec: 130.852
INFO:tensorflow:loss = 0.3313904, step = 201 (0.764 sec)
INFO:tensorflow:global_step/sec: 135.722
INFO:tensorflow:loss = 0.08310337, step = 301 (0.737 sec)
INFO:tensorflow:global_step/sec: 139.879
INFO:tensorflow:loss = 0.13494043, step = 401 (0.715 sec)
INFO:tensorflow:global_step/sec: 144.19
INFO:tensorflow:loss = 0.06767261, step = 501 (0.694 sec)
INFO:tensorflow:global_step/sec: 147.938
INFO:tensorflow:loss = 0.09191691, step = 601 (0.676 sec)
INFO:tensorflow:global_step/sec: 127.188
INFO:tensorflow:loss = 0.15389094, step = 701 (0.786 sec)
INFO:tensorflow:global_step/sec: 122.919
INFO:tensorflow:loss = 0.07212783, step = 801 (0.814 sec)
INFO:tensorflow:global_step/sec: 123.735
INFO:tensorflow:loss = 0.17153545, step = 901 (0.811 sec)
INFO:tensorflow:global_step/sec: 125.078
INFO:tensorflow:loss = 0.20417586, step = 1001 (0.797 sec)
INFO:tensorflow:Saving checkpoints for 1010 into ./trained_model/model.ckpt.
INFO:tensorflow:Loss for final step: 0.030963236.
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
WARNING:tensorflow:Casting <dtype: 'float32'> labels to bool.
INFO:tensorflow:Starting evaluation at 2018-04-01-14:20:58
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from ./trained_model/model.ckpt-1010
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Evaluation [10/100]
INFO:tensorflow:Evaluation [20/100]
INFO:tensorflow:Finished evaluation at 2018-04-01-14:21:00
INFO:tensorflow:Saving dict for global step 1010: accuracy = 0.8775367, accuracy/baseline_label_mean = 0.83344996, accuracy/threshold_0.700000_mean = 0.8353494, auc = 0.9727974, auc_precision_recall = 0.9941381, global_step = 1010, labels/actual_label_mean = 0.83344996, labels/prediction_mean = 0.7044494, loss = 0.34910947, precision/positive_threshold_0.700000_mean = 0.9930719, recall/positive_threshold_0.700000_mean = 0.8080844, rmse = 0.30576178, training/hptuning/metric = 0.30576178
INFO:tensorflow:Restoring parameters from ./trained_model/model.ckpt-1010
INFO:tensorflow:Assets added to graph.
INFO:tensorflow:No assets to write.
INFO:tensorflow:SavedModel written to: ./trained_model/export/Servo/temp-1522592460/saved_model.pb

**Comment by lakshmanok** [comment]
This is because --batch_size in task.py defaults to 100, not 512.

**Comment by mshearer0** [comment]
Thanks, that works.

So its correct to specify  fix_global_step_increment_bug=True  ?
